src/core/thread/mle.cpp

```diff
@@ -1248,8 +1252,9 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
     uint8_t                  length      = 0;
     uint8_t                  counter     = 0;
     uint16_t                 startOffset = aMessage.GetLength();
-    uint8_t                  iterator    = 0; // used to iterate external multicast addresses.
-    Ip6::Address             address;
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    Ip6::Address address;
```

Comment:
nit: rename to `dua` or `domainUnicastAddress`?

Reply:
Good suggestion. Thanks. Changed in new pushed commit.